include/cpp-sort/adapters/hybrid_adapter.h

```diff
@@ -122,23 +122,52 @@ namespace cppsort
                     is_stable<Sorter(Args...)>
                 >;
         };
-    }
 
-    ////////////////////////////////////////////////////////////
-    // Adapter
+        template <template <class...> class Flattenable, class TypeList, class Accumulator>
+        struct flatten_fold;
 
-    template<typename... Sorters>
-    struct hybrid_adapter:
-        public detail::check_iterator_category<Sorters...>,
-        public detail::check_is_always_stable<Sorters...>,
-        public sorter_facade_fptr<hybrid_adapter<Sorters...>>
-    {
+        template <
+                template <class...> class Flattenable,
+                template <class...> class TsList, class Front, class... Rest,
+                template <class...> class AsList, class... As
+        >
+        struct flatten_fold<Flattenable, TsList<Front, Rest...>, AsList<As...>>
+        {
+            using type = typename flatten_fold<Flattenable, TsList<Rest...>, AsList<As..., Front>>::type;
+        };
+
+        template <
+                template <class...> class Flattenable,
+                template <class...> class TsList, class... InnerTs, class... Rest,
+                template <class...> class AsList, class... As
+        >
+        struct flatten_fold<Flattenable, TsList<Flattenable<InnerTs...>, Rest...>, AsList<As...>>
+        {
+            using type = typename flatten_fold<Flattenable, TsList<InnerTs..., Rest...>, AsList<As...>>::type;
+        };
+
+        template <
+                template <class...> class Flattenable,
+                template <class...> class TsList,
+                class Accumulator
+        >
+        struct flatten_fold<Flattenable, TsList<>, Accumulator>
+        {
+            using type = Accumulator;
+        };
+
+        template<typename... Sorters>
+        struct hybrid_adapter_impl:
+                public detail::check_iterator_category<Sorters...>,
+                public detail::check_is_always_stable<Sorters...>,
+                public sorter_facade_fptr<hybrid_adapter_impl<Sorters...>>
+        {
         public:
 
-            hybrid_adapter() = default;
+            hybrid_adapter_impl() = default;
 
             // Automatic deduction guide
-            constexpr explicit hybrid_adapter(Sorters...) noexcept {}
+            constexpr explicit hybrid_adapter_impl(Sorters...) noexcept {}
```

Comment:
That constructor probably won't be used as a deduction guide since it's only part of the implementation, I think we can get rid of it/